packages/query-core/src/__tests__/queryObserver.test.tsx

```diff
@@ -52,6 +52,51 @@ describe('queryObserver', () => {
     unsubscribe()
   })
 
+  test('Should not refetch when enabled is a callback and returns false', async () => {
```

Comment:
can we add a test please that checks if the query runs when enabled is a function that returns false?

Reply:
Updated and added a few more tests.